browser/components/newtab/NewTabPrefsProvider.jsm

```diff
@@ -31,7 +31,24 @@ PrefsProvider.prototype = {
   observe(subject, topic, data) { // jshint ignore:line
     if (topic === "nsPref:changed") {
       if (prefsSet.has(data)) {
-        this.emit(data);
+        switch (data) {
+          case "browser.newtabpage.enabled":
+            this.emit("setEnabled", {
```

Comment:
Can you please use the name of the pref as the message it emits?

It makes for good semantics: you listen "on" a pref

i.e. you'd do NewTabPrefsProvider.on("browser.newtabpage.enabled") as opposed to NewTabPrefsProvider.on("setEnabled")